DOCUMENTATION.md

```diff
@@ -849,6 +893,239 @@ which serves to add tasks for Chainlink Keepers to perform on client contracts.
+
+> **Note**
+> The following are methods that are available only in HRE
+
+#### Initialize SubscriptionManager class
+- **Method:** initializeFunctionsSubscriptionManager
+- **Description:** Initialize [SubscriptionManager class](https://github.com/smartcontractkit/functions-toolkit#functions-billing-subscription-management)
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+  - `linkTokenAddress`: Address of Link Token
+
+#### Initialize ResponseListener class
+- **Method:** initializeFunctionsResponseListener
+- **Description:** Initialize [ResponseListener class](https://github.com/smartcontractkit/functions-toolkit#functions-response-listener)
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+
+#### Initialize SecretsManager class
+- **Method:** initializeFunctionsSecretsManager
+- **Description:** Initialize [SecretsManager class](https://github.com/smartcontractkit/functions-toolkit#functions-encrypted-secrets-management)
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+  - `donId`: ID of the DON to work with
+
+#### Listen for a response for single Functions request
+- **Method:** listenForResponse
+- **Description:** Listen for a response for single Functions request
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+  - `requestId`: ID of the DON to which the Functions request will be sent
+  - `timeout`: Subscription ID (Default: 300_000)
+
+#### Listen for a response for single Functions request (from transaction)
+- **Method:** listenForResponseFromTransaction
+- **Description:** Listen for a response for single Functions request (from transaction)
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+  - `transactionHash`: Transaction hash to start listen from
+  - `timeout`: Listener timeout (Default: 300000)
+  - `confirmations`: Number of block confirmations (Default: 2)
+  - `checkInterval`: Frequency of checking for a response to appear on-chain (Default: 2000)
+
+#### Listen for responses
+- **Method:** listenForResponses
+- **Description:** Listen for multiple Functions responses
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+  - `subscriptionId`: Subscription ID
+  - `callback`: Callback to be called when responses are fulfilled
+
+#### Stop responses listener
+- **Method:** stopListeningForResponses
+- **Description:** Stop responses listener
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+
+#### Stop responses listener
+- **Method:** stopListeningForResponses
+- **Description:** Stop responses listener
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+
+#### Fetch DON public keys
+- **Method:** fetchKeys
+- **Description:** Fetch DON public keys
+- **Arguments:**
+  - `functionsRouterAddress`: Address of Functions Router
+  - `donId`: ID of the DON to fetch keys from
+
+#### Encrypt Gist URLs
```

Comment:
Since off-chain user-hosted secrets can be on any public url (not just gists) we should replace reference to gists with "off-chain secrets" URLs
This also keeps the phrasing consistent with the NPM package docs etc.